src/libs/Violations/ViolationsUtils.ts

```diff
@@ -104,7 +104,7 @@ const ViolationsUtils = {
             case 'billableExpense':
                 return translate('violations.billableExpense');
             case 'cashExpenseWithNoReceipt':
-                return translate('violations.cashExpenseWithNoReceipt', {amount: violation.data?.amount ?? ''});
+                return translate('violations.cashExpenseWithNoReceipt', violation.data ?? undefined);
```

Comment:
Out of curiosity, given data is optional (and can be null), why is it necessary to add the fallback to `undefined`? is it because of the type of the second param of `translate`?

Reply:
The function for this translation is:

```
cashExpenseWithNoReceipt: ({formattedLimit}: ViolationsCashExpenseWithNoReceiptParams = {}) => `Receipt required${formattedLimit ? ` over ${formattedLimit}` : ''}`,
```

Which is pretty verbose, so let's take a simpler version

```
myFunction: ({prop}:ParamType = {}) => prop
```

This function takes a single `ParamType` object as a prop and destructures to get the value of prop.

If the anonymous param object is `undefined` then it uses the default value `{}`, and then gets the value of `{}.prop` which is `undefined`

If we pass `null` as the argument it will be used and the code will try to evaluate `null.prop` and throw an error like `TypeError: Cannot read property 'prop' of null`

So we pass undefined to let the function use the default value, which makes `prop` undefined.

I was going back and forth on where to hide the complexity.  In the end I thought it was better to keep the consistency in the translations file, where all the functions are one-liners.

The alternative would be to have something like:
```
cashExpenseWithNoReceipt: (params: ViolationsCashExpenseWithNoReceiptParams | null) => {
    const {formattedLimit} = params ?? {}
    return `Receipt required${formattedLimit ? ` over ${formattedLimit}` : ''
}
```
But that doesn't match the rest of the functions in the translation files, which generally are one liners without logic in there.

I could also define the ViolationsCashExpenseWithNoReceiptParams to be nullable, but again that would be out of line with the rest of the translations, and my general preference in giant files like the translations and their param types is to just be the same as the other code in that file.

Reply:
This is much cleaner now -- I removed the default case and handle the initialization at the top of the function for all variables at once. 

So now it just looks like this:

`return translate('violations.cashExpenseWithNoReceipt', {formattedLimit});`

Reply:
>If the anonymous param object is undefined then it uses the default value {},

Oh right (cause passing undefined is the same as not passing anything), I didn't connect these dots!